test/integration/bundle/cluster_bundle_validation_test.go

```diff
@@ -0,0 +1,275 @@
+							metadataField = "labels"
+						}
+						matchErr := "spec.target.%s.metadata.%s: Forbidden: must not use forbidden domains as prefixes (e.g., trust-manager.io)"
+						Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field, metadataField)))
+					} else {
+						Expect(cl.Create(ctx, bundle)).To(Succeed())
+					}
+				},
+				Entry("when trust-manager.io annotations are used", &trustmanagerapi.TargetMetadata{Annotations: map[string]string{"trust-manager.io/hash": "test"}}, true),
+				Entry("when trust-manager.io labels are used", &trustmanagerapi.TargetMetadata{Labels: map[string]string{"trust-manager.io/bundle": "bundle"}}, true),
+				Entry("when non-reserved annotations are used", &trustmanagerapi.TargetMetadata{Annotations: map[string]string{"not-trust-manager.io/hash": "test"}}, false),
+				Entry("when non-reserved labels are used", &trustmanagerapi.TargetMetadata{Labels: map[string]string{"not-trust-manager.io/bundle": "bundle"}}, false),
+			)
+
+			DescribeTable("should validate format",
+				func(format trustmanagerapi.BundleFormat, matchErr string) {
+					selectorAccessor(&trustmanagerapi.KeyValueTarget{
+						Data: []trustmanagerapi.TargetKeyValue{{
+							Key:    "ca-bundle",
+							Format: format,
+						}},
+					})
+					if matchErr != "" {
+						Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field)))
+					} else {
+						Expect(cl.Create(ctx, bundle)).To(Succeed())
+					}
+				},
+				Entry("when unset", trustmanagerapi.BundleFormat(""), ""),
+				Entry("when PEM", trustmanagerapi.BundleFormatPEM, ""),
+				Entry("when PKCS12", trustmanagerapi.BundleFormatPKCS12, ""),
+				Entry("when unknown", trustmanagerapi.BundleFormat("JKS"), "spec.target.%s.data[0].format: Unsupported value: \"JKS\": supported values: \"PEM\", \"PKCS12\""),
+			)
+
+			DescribeTable("should validate fields reserved for PCKS12 format",
+				func(pkcs12 trustmanagerapi.PKCS12) {
+					targetKeyValue := trustmanagerapi.TargetKeyValue{
+						Key:    "ca-bundle",
+						PKCS12: pkcs12,
+					}
+					selectorAccessor(&trustmanagerapi.KeyValueTarget{
+						Data: []trustmanagerapi.TargetKeyValue{targetKeyValue},
+					})
+					var pkcs12Field = "password"
+					if pkcs12.Profile != "" {
+						pkcs12Field = "profile"
+					}
+					matchErr := "spec.target.%s.data[0].%s: Forbidden: may only be set when format is 'PKCS12'"
+					Expect(cl.Create(ctx, bundle)).Should(MatchError(ContainSubstring(matchErr, field, pkcs12Field)))
+
+					targetKeyValue.Format = trustmanagerapi.BundleFormatPKCS12
+					selectorAccessor(&trustmanagerapi.KeyValueTarget{
+						Data: []trustmanagerapi.TargetKeyValue{targetKeyValue},
+					})
+					Expect(cl.Create(ctx, bundle)).To(Succeed())
+				},
+				Entry("when password set", trustmanagerapi.PKCS12{Password: ptr.To("my-password")}),
+				Entry("when profile set", trustmanagerapi.PKCS12{Profile: trustmanagerapi.Modern2023PKCS12Profile}),
```

Comment:
question: are we missing tests for when profile != PKCS12 but "password" is provided? I might be missing something